src/tools/run-make-support/src/command.rs

```diff
@@ -388,9 +388,15 @@ impl CompletedProcess {
         self
     }
 
+    /// Check the **exit status** of the process. On Unix, this is *not* the **wait status**.
+    ///
+    /// See [`std::process::ExitStatus::code`]. This is not to be confused with
+    /// [`std::process::ExitCode`].
     #[track_caller]
     pub fn assert_exit_code(&self, code: i32) -> &Self {
-        assert!(self.output.status.code() == Some(code));
+        // FIXME(jieyouxu): this should really be named `exit_status`, because std has an `ExitCode`
+        // that means a different thing.
```

Comment:
`ExitStatus` is effectively an enum between an exit code and information about the signal that caused the process to exit. This function checks that the `ExitStatus` is an exit code and in particular that it is the given exit code.

Reply:
Yeah, this FIXME was outdated from debugging, I'll send a follow-up to remove it (the asserion behaves as intended)

Reply:
Nvm, fixed